users/vieting/experiments/switchboard/transducer/feat/experiments.py

```diff
@@ -277,6 +287,142 @@ def run_rasr_gt_stage1():
+                                    "/conv_h_filter/conv_h_filter": "features/conv_h_filter/conv_h_filter",
+                                    "/conv_l/W": "features/conv_l/W",
+                                    "/conv_l_act/bias": "features/conv_l_act/bias",
+                                    "/conv_l_act/scale": "features/conv_l_act/scale"
+                                },
+                            }
+                        },
+                    },
+                },
+                report_args={"alignment": "ctc-conf-e400"},
+                lr_args={"dynamic_learning_rate": dynamic_learning_rate_configurable},
+                feature_args=feature_args,
+            ),
+            "bs15k_align-ctc-conf-e400_feat-ctc-e400_froozen": dict(
+                returnn_args={
+                    **returnn_args,
+                    "extra_args": {
+                        # data sequence is longer by factor 4 because of subsampling and 80 because of feature extraction vs.
+                        # raw waveform
+                        "chunking": ({"classes": 64, "data": 64 * 4 * 80}, {"classes": 32, "data": 32 * 4 * 80}),
+                        "gradient_clip": 20.0,
+                        "learning_rate_control_error_measure": "sum_dev_score",
+                        "min_learning_rate": 1e-6,
+                        "preload_from_files": {
+                            "existing-model": {
+                                "filename": nn_system_ctc.train_jobs["conformer_bs2x5k_scf_baseline_preemphasis97_wn"].out_checkpoints[400],
+                                "init_for_train": True,
+                                "prefix": "features", 
+                                "var_name_mapping": {
+                                    "/conv_h_filter/conv_h_filter": "features/conv_h_filter/conv_h_filter",
+                                    "/conv_l/W": "features/conv_l/W",
+                                    "/conv_l_act/bias": "features/conv_l_act/bias",
+                                    "/conv_l_act/scale": "features/conv_l_act/scale"
+                                },
+                            }
+                        },
+                    },
+                    "staged_opts": {1: "freeze_features"},
+                },
+                report_args={"alignment": "ctc-conf-e400"},
+                lr_args={"dynamic_learning_rate": dynamic_learning_rate_configurable},
+                feature_args=feature_args,
+            ),
+        },
+        num_epochs=300,
+        evaluation_epochs=[270, 280, 290, 300],
+        prefix="viterbi_scf_",
+    )
+    config = copy.deepcopy(nn_args.returnn_recognition_configs["viterbi_scf_bs15k_align-ctc-conf-e400"].config)
+    config["extern_data"]["data"]["dtype"] = "float32"
+    config["extern_data"]["classes"]["dtype"] = "int32"
+    nn_args.returnn_recognition_configs["viterbi_scf_bs15k_align-ctc-conf-e400"].config = config
```

Comment:
Now you only change this for one of the three trainings. It would be better to do this automatically, see my previous comment.

Reply:
Why do the training configs need different data types at all?

Reply:
I think because the input is from a RETURNN dataset in training and from RASR in recognition where the types differ.

Reply:
I just pushed something to main, please rebase and adjust your PR accordingly.

Reply:
I did. Is this related to the extern data? Do I keep them as they are or change them in baseline_args.py for all trainings?

Reply:
It should work without this hack here now.